com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateManager.cs

```diff
@@ -4,216 +4,193 @@
+        /// <returns></returns>
+        private static bool RemoveSystem(ref PlayerLoopSystem system, PlayerLoopSystem toRemove)
         {
-            if (CurrentNetworkUpdateEngine == null)
+            if (system.subSystemList == null)
             {
-                CurrentNetworkUpdateEngine = new InternalNetworkUpdateEngine();
+                return false;
             }
 
-            switch (updateStage)
+            for (int i = 0; i < system.subSystemList.Length; ++i)
             {
-                case NetworkUpdateStages.PREUPDATE:
-                {
-                    CurrentNetworkUpdateEngine.PreUpdateRegister(updateAction);
-                    break;
-                }
-                case NetworkUpdateStages.FIXEDUPDATE:
+                if (system.subSystemList[i].type == toRemove.type)
                 {
-                    CurrentNetworkUpdateEngine.FixedUpdateRegister(updateAction);
-                    break;
-                }
-                case NetworkUpdateStages.UPDATE:
-                {
-                    CurrentNetworkUpdateEngine.UpdateRegister(updateAction);
-                    break;
+                    RemoveSystemAt(ref system, toRemove);
+                    return true;
                 }
-                case NetworkUpdateStages.LATEUPDATE:
+            }
+
+            for (var i = 0; i < system.subSystemList.Length; i++)
+            {
+                if (RemoveSystem(ref system.subSystemList[i], toRemove))
                 {
-                    CurrentNetworkUpdateEngine.PostUpdateRegister(updateAction);
-                    break;
+                    return true;
                 }
             }
+
+            return false;
         }
 
         /// <summary>
-        /// AppStart
-        /// Initial definition of the four primary network update stages
+        /// RemoveSystemAt
+        /// Copies the subSystemList of the given system, but excludes the specified PlayerLoopSystem from the list
         /// </summary>
-        [RuntimeInitializeOnLoadMethod]
-        private static void AppStart()
+        /// <param name="system">PlayerLoopSystem to be inserted into</param>
+        /// <param name="toRemove">PlayerLoopSystem to insert</param>
+        private static void RemoveSystemAt(ref PlayerLoopSystem system, PlayerLoopSystem toRemove)
         {
-            var def = PlayerLoop.GetCurrentPlayerLoop();
-
-            if (CurrentNetworkUpdateEngine == null)
+            PlayerLoopSystem[] newSubSystems = new PlayerLoopSystem[system.subSystemList.Length-1];
+            for (int i = 0, newSystemIdx = 0; i < newSubSystems.Length-1; ++i)
             {
-                CurrentNetworkUpdateEngine = new InternalNetworkUpdateEngine();
+                if ( system.subSystemList[i].type != toRemove.type && system.subSystemList[i].updateFunction != toRemove.updateFunction )
+                {
+                    newSubSystems[newSystemIdx++] =  system.subSystemList[i];
```

Comment:
IDK but just wanted to double-check:
`newSubSystems[newSystemIdx++]` is post-increment intentional here or should it be pre-?

Reply:
Yes, that is intentional.  Once the current SubSystem has been assigned, increment to the next position in the new SubSystem array to be applied to the PlayerLoop root PlayerLoopSystem.